docs/user-guide/experimental-api.md

```diff
@@ -0,0 +1,30 @@
+# Experimental API
+
+## What is the experimental API
+
+The exerimental API is an API group indended for experimenting with and testing
+APIs. To make this possible, unlike other API groups, there is no guarantee of
+future support, compatibility, or even existence. We do this to lower the
+development costs around making new APIs, to let experiments proceed without
+ongoing maintenance costs.
+
+## Where is the experimental API group?
+
+It is hosted by kube-apiserver, at /apis/experimental/v1alpha1. `kubectl`
+understands experimental objects, so they will work just like other objects in
+the Kubernetes system. You'll have to explicitly say "experimental/" in front of
+resource names, e.g. "experimental/daemonset". Because of the somewhat ephemeral
```

Comment:
@derekwaynecarr, in that case it will call VersionAndKindForResource("deamonset") to get a version, which searches "daemonset" in both "v1" and "experimental/v1alpha1", and will find "daemonset" in "experimental/v1alpha1". Then, "experimental/v1alpha1" is passed into NegotiateVersion().

Currently NegotiateVersion, client.Config, etc don't work properly with groups. #14592 #14941 #14951 try to fix them. We'll focus on those PRs after 1.1.